### PR TITLE
fix(dotenv): introduce safe parsing of .env files

### PR DESCRIPTION
**Pull request by mcornella**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] If I used AI tools (ChatGPT, Claude, Gemini, etc.) to assist with this contribution, I've disclosed it below.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- fix(dotenv): expect explicit yes before loading .env file
- fix(dotenv): implement secure parsing for .env files and add comprehensive tests
- feat(dotenv): check for .env file size to prevent DoS
- fix(dotenv): forbid setting special variables
- fix(dotenv): FIFO shouldn't be read twice
- fix(dotenv): unknown vars should expand to empty
- fix(dotenv): reject extremely large named pipes
- docs(dotenv): update to new parsing system
- fix(dotenv): add support for escaped dollars
- chore(dotenv): only declare local variables once

## AI disclosure

We've had AI assistance in building test cases and catching edge cases. The choosing of zsh features to safely parse .env files has been mostly human work. Also we've manually reviewed and verified all code.

### CODE REVIEW

**Review by carlosala — DISMISSED** [review]

File: plugins/dotenv/tests/security.zunit [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+GIT_TEMPLATE_DIR=./malicious_templates # for persistence
+
+# Special exported variables
+PATH=./malicious_bin:$PATH
+EDITOR=./malicious
+VISUAL=./malicious
+PAGER=./malicious
+EOF
+)
+
```
[the remark this answers is not in the document]
Comment (mcornella) [reply plugins/dotenv/tests/security.zunit:207]: Already tested elsewhere

File: plugins/dotenv/dotenv.plugin.zsh [hunk 2]
```diff
@@ -7,9 +7,271 @@
...
+    parsed_vars[$key]="$value"
+
+    # Normal mode: export the variable
+    if [[ "$mode" == "export" ]]; then
+      typeset -x "$key"="$value"
+    fi
+  done
+
+  # In test mode, set DOTENV_TEST_VARS
+  typeset -gA DOTENV_TEST_VARS
```
[the remark this answers is not in the document]
Comment (mcornella) [reply plugins/dotenv/dotenv.plugin.zsh:203]: Dumb

**Review by carlosala — APPROVED** [review]

**Review by carlosala — DISMISSED** [review]

**Review by carlosala — APPROVED** [review]

**Review by carlosala — DISMISSED** [review]

**Review by carlosala — APPROVED** [review]

**Comment by mcornella** [comment]
This PR is related to [this advisory](https://github.com/ohmyzsh/ohmyzsh/security/advisories/GHSA-3rgh-p3mg-rjqg).
